lib/units/auth/oauth2/index.js

```diff
@@ -46,8 +56,9 @@ module.exports = function(options) {
         }))
       }
       else {
-        log.warn('Missing email in profile', req.user)
-        res.redirect('/auth/oauth/')
+        log.warn('Missing or disallowed email in profile', req.user)
+        res.send('<html><body>Missing or rejected email address ' +
```

Comment:
I would like to move this to a template, it'd make it easier to improve it later.